### ISSUE TITLE
Typescript inconsistencies when using useTranslation hook.

### ISSUE THREAD

**Issue by ImADrafter** [i18next/react-i18next#1277]
First of all, TY for the amazing feature of adding type-safety to i18n t calls. It is an improvement for sure!

## 🐛 Bug Report

Using the `useTranslation` hook, I found some inconsistencies between how the actual `t` function works in javascript and its typescript signature.

I think its mainly caused for how the hook signature manages namespaces. Please take a look at the CodeSandbox bellow.

What I think the expected behaviour should be is, if the function actually translates, TS should't complain, and viceversa. Some examples may seem obvious, but please pay attention to those where:

Compiles: ✅  Translates: ❌ - In this case, it compiles, but it should not
Compiles: ❌  Translates: ✅ - Should compile, since the function actually translates

## To Reproduce

[CodeSandbox](https://codesandbox.io/s/react-i18next-1268-forked-1gxzv?file=/src/App.tsx)

Related issue: https://github.com/i18next/react-i18next/issues/1268 -> Same problem but with fallback NS

## Your Environment

- *i18next version*: i.e. 19.9.2
- *react-i18next version*: i.e. 11.8.10

**Comment by pedrodurek** [comment]
Hey @ImADrafter, thank you for your feedback!
I checked your example and here is my outcome:
* ExampleWithDefaults:
It was working before, so it's probably something that we've changed recently, I'll **investigate** it.

* Other examples:
It's working as expected. When relying on typescript, the accepted patterns will be:

```tsx
const {t} = useTranslation();
t("c"); // Compile
t("translation:c"); // Won't compile
```
```tsx
const {t} = useTranslation('ns1');
t("a"); // Compile
t("ns1:a"); // Won't compile
```
```tsx
const {t} = useTranslation(['ns1', 'ns2']);
t("ns1:a"); // Compile
t("ns2:b"); // Compile
t("a"); // Won't compile
```
The type inferrance was designed like that in order to improve compilation performance, you may not noticed in small projects, but in mid and large projects, it makes a huge difference. 
In a nutshell, since `a` and `ns1:a` perform the same action in a plain format, I'm accepting just `a` to reduce the number of iterations to generate the inferred keys. The same rule applies for array formats.
In my tests, it reduced the compilation time in around 60~70%.

**Comment by pedrodurek** [comment]
Hey @ImADrafter, It should be fixed now!

**Comment by ImADrafter** [comment]
> Hey @ImADrafter, It should be fixed now!

I checked on the sandbox, and you are right! TY for taking the time on this! 🎉 

> In a nutshell, since a and ns1:a perform the same action in a plain format, I'm accepting just a to reduce the number of iterations to generate the inferred keys. The same rule applies to array formats.
In my tests, it reduced the compilation time by around 60~70%.

It makes sense that compilation time could dramatically increase in large projects, so I think this is the best approach :) TY again

**Comment by vricop** [comment]
> Hey @ImADrafter, thank you for your feedback! I checked your example and here is my outcome:
> 
>     * ExampleWithDefaults:
>       It was working before, so it's probably something that we've changed recently, I'll **investigate** it.
> 
>     * Other examples:
>       It's working as expected. When relying on typescript, the accepted patterns will be:
> 
> 
> ```tsx
> const {t} = useTranslation();
> t("c"); // Compile
> t("translation:c"); // Won't compile
> ```
> 
> ```tsx
> const {t} = useTranslation('ns1');
> t("a"); // Compile
> t("ns1:a"); // Won't compile
> ```
> 
> ```tsx
> const {t} = useTranslation(['ns1', 'ns2']);
> t("ns1:a"); // Compile
> t("ns2:b"); // Compile
> t("a"); // Won't compile
> ```
> 
> The type inference was designed like that in order to improve compilation performance, you may not notice it in small projects, but in mid and large projects, it makes a huge difference. In a nutshell, since `a` and `ns1:a` perform the same action in a plain format, I'm accepting just `a` to reduce the number of iterations to generate the inferred keys. The same rule applies to array formats. In my tests, it reduced the compilation time by around 60~70%.
> 
> Another advantage is that it'll standardize the way people use the library.

This isn't reflected in the docs, which makes things confusing. This is what the docs state:

```js
// load a specific namespace
// the t function will be set to that namespace as default
const { t, i18n } = useTranslation('ns1');
t('key'); // will be looked up from namespace ns1

// load multiple namespaces
// the t function will be set to first namespace as default
const { t, i18n } = useTranslation(['ns1', 'ns2', 'ns3']);
t('key'); // will be looked up from namespace ns1
t('key', { ns: 'ns2' }); // will be looked up from namespace ns2
```
